src/packages/external-event/external-event.ts

```diff
@@ -0,0 +1,136 @@
+    }
+  } catch (error) {
+    logger.error(
+      `POST /uploadExternalEventType: Error occurred during External Event Type ${external_event_type_name} upload`,
+    );
+    logger.error((error as Error).message);
+    res.status(500).send({ message: (error as Error).message });
+    return;
+  }
+
+  logger.info(`POST /uploadExternalEventType: Attribute schema was VALID`);
+
+  // Run the Hasura migration for creating an external event
+  const externalEventTypeInsertInput: ExternalEventTypeInsertInput = {
+    attribute_schema: attribute_schema,
+    name: external_event_type_name,
+  };
+
+  const response = await fetch(GQL_API_URL, {
+    body: JSON.stringify({
+      query: gql.CREATE_EXTERNAL_EVENT_TYPE,
+      variables: { eventType: externalEventTypeInsertInput },
+    }),
+    headers,
+    method: 'POST',
+  });
+
+  type CreateExternalEventTypeResponse = {
+    data: { createExternalEventType: { attribute_schema: object; name: string } | null };
+  };
+  const jsonResponse = await response.json();
+  const createExternalEventTypeResponse = jsonResponse as CreateExternalEventTypeResponse | HasuraError;
+
+  res.json(createExternalEventTypeResponse);
+}
+
+export default (app: Express) => {
+  /**
+   * @swagger
+   * /uploadExternalEventType:
+   *   post:
+   *     security:
+   *       - bearerAuth: []
+   *     consumes:
+   *       - application/json
+   *     produces:
+   *       - application/json
+   *     parameters:
+   *       - in: header
+   *         name: x-hasura-role
+   *         schema:
+   *           type: string
+   *           required: false
+   *     requestBody:
+   *       content:
+   *         application/json:
+   *           schema:
+   *             type: object
+   *             properties:
+   *               attribute_schema:
+   *                 type: object
+   *               external_event_type_name:
+   *                 type: string
+   *             required:
+   *               - external_event_type_name
+   *                 attribute_schema
+   *     responses:
+   *       200:
+   *         description: Created External Event Type
+   *         content:
+   *           application/json:
+   *             schema:
+   *                properties:
+   *                  attribute_schema:
+   *                    description: JSON Schema for the created External Event Type's attributes
+   *                    type: object
+   *                  name:
+   *                    description: Name of the created External Event Type
+   *                    type: string
+   *       403:
+   *         description: Unauthorized error
+   *       401:
+   *         description: Unauthenticated error
+   *     summary: Uploads an External Event Type definition (containing name & attributes schema) to Hasura.
+   *     tags:
+   *       - Hasura
+   */
+  app.post('/uploadExternalEventType', uploadExternalEventType);
```

Comment:
This endpoint is currently unauthenticated. Example of an authenticated endpoint from L716 of `packages/plan/plan.ts`. Please also speak with @duranb regarding the `upload.single` part.

```ts
app.post('/uploadDataset', upload.single('external_dataset'), refreshLimiter, auth, uploadDataset);
```

Reply:
Re-worked the routes so they properly use `auth` with `plan.ts` as a reference